src/web/js-interop/index.md

```diff
@@ -0,0 +1,125 @@
+---
+title: JavaScript interoperability
+short-title: JS interop
+description: Integrate JavaScript code into your Dart web app.
+---
+
+The [Dart web platform](/overview#web-platform) supports communication with
+JavaScript apps and libraries, and browser DOM APIs, using the `js_interop`
+package, also known as `dart:js_interop`.
+Web developers can benefit from using external JS libraries in their Dart code, without
+having to rewrite anything in Dart.
+
+## Static interop
+
+The principal model of JS interop in Dart is **static interop**.
+Static interop means interop that requires staticly typing external members,
+rather than allowing dynamic invocations.
+This enables [features](#features) like better performance, type soundness, and more.
+
+Performant static interop model is made possible by **[inline classes][]**.
+Inline classes are a special class type that wrap an existing type into a new static type,
+without the overhead of a traditional class, enabling zero cost wrapping.
+
+### Why static interop?
+
+Static interop represents the desire to decouple Dart's "core" from the platform
+where it's embedded
+(*maybe definition of "embedded" or examples here, like the browser, Flutter, etc.?*).
+
+Our [previous iterations of JS interop][] provided the capabillity
+to access the embedder, but were not primed to handle that decoupling.
+Too much of the process was entwined into the Dart side,
+like writing all the bindings to the browser
+(*not sure if I understood that note correctly*). 
+They also had limitations around using the DOM,
+typing (which is a cornerstone of Dart),
+and expanding to new interop types
+(*this is me trying to refer to wasm without actually saying, idk if "interop types" is the right term*). 
+
+Static interop addresses all of the shortcomings of Dart's previous JS interop solutions.
+Check out the [Features](#features) section for a summary of improvements. 
+
+[previous iterations of JS interop]: /web/js-interop/past-js-interop
+
+## Usage
```

Comment:
This usage section is like a quick-start, but what do you think about linking to a more elaborate tutorial about basic usage, @srujzs? It might be "core" to the new JS docs to have a single page under `/js-interop/tutorials` that walks through the "Exposing..." and "Consuming..." portions of your design doc, which to me seem like the "basics".

There will probably be some overlap if we do create a longer basics tutorial, but the difference in presentation might warrant some conceptual duplication. Like, to me, this section in the overview is just like "1. Do this, 2. Do this, 3. And now it works", whereas a dedicated usage tutorial can elaborate more, like "1. First you start by doing this because ...., and it works like ..., and you have a few options on how to approach ... depending on .... 2. Now that you've ..., you can ...., because you needed to .... first before you could ....", etc. 

I'm definitely a little blind to how much there even is to say about basic usage, so maybe the basic "steps" are actually so simple that they can be covered in a handful of steps and there's nothing to elaborate? But it might be a good place to cover other "basic" parts in a guided teaching style. Like I can imagine when you're talking about consuming JS, you could make up an arbitrary type conversion section to go over, and talk about using extension types a little more, etc.

So this might not be necessary, just putting it out there in case it makes sense! And if you like the idea I can draft something up for what that tutorial page might look like.